packages/ckeditor5-engine/tests/view/placeholder.js

```diff
@@ -435,6 +435,131 @@ describe( 'placeholder', () => {
 			expect( viewRoot.getChild( 0 ).hasAttribute( 'data-placeholder' ) ).to.be.true;
 			expect( viewRoot.getChild( 0 ).hasClass( 'ck-placeholder' ) ).to.be.false;
 		} );
+
+		describe( 'aria', () => {
+			it( 'should set `aria-placeholder` attribute to nearest element with `contenteditable=true`', () => {
+				setData( view, '<div contenteditable="true"><div>{another div}</div></div>' );
+
+				const parent = viewRoot.getChild( 0 );
+				const child = parent.getChild( 0 );
+
+				child.placeholder = 'foo bar';
+				enablePlaceholder( {
+					view,
+					element: child
+				} );
+
+				expect( parent.getAttribute( 'aria-placeholder' ) ).to.be.equal( 'foo bar' );
+			} );
+
+			it( 'should not set `aria-placeholder` attribute to nearest element with `contenteditable=false`', () => {
+				setData( view, '<div contenteditable="false"><div>{another div}</div></div>' );
+
+				const parent = viewRoot.getChild( 0 );
+				const child = parent.getChild( 0 );
+
+				child.placeholder = 'foo bar';
+				enablePlaceholder( {
+					view,
+					element: child
+				} );
+
+				expect( parent.getAttribute( 'aria-placeholder' ) ).to.be.undefined;
+			} );
+
+			it( 'should join placeholders to `aria-placeholder`', () => {
+				setData(
+					view,
+					'<div contenteditable="true">' +
+						'<div>{another div 1}</div>' +
+						'<div>{another div 2}</div>' +
+					'</div>'
+				);
+
+				const parent = viewRoot.getChild( 0 );
+				const child1 = parent.getChild( 0 );
+				const child2 = parent.getChild( 1 );
+
+				child1.placeholder = ' Hello World...! ';
+				child2.placeholder = 'Helloooo!!... ';
+
+				enablePlaceholder( {
+					view,
+					element: child1
+				} );
+
+				enablePlaceholder( {
+					view,
+					element: child2
+				} );
+
+				expect( parent.getAttribute( 'aria-placeholder' ) ).to.be.equal( 'Hello World!, Helloooo!!' );
```

Comment:
What value would that be to the users? There could be 10 placeholders in 10 random places in the document (at different nesting levels). It doesn't seem like aggregation is the way to go here. Here's the screenshot from http://fake.ckeditor.com:8125/ckeditor5-heading/tests/manual/title.html

<img width="1880" alt="Screenshot 2024-04-29 at 16 09 09" src="https://github.com/ckeditor/ckeditor5/assets/1099479/681382c7-2b93-416e-96ea-1a90fe7f813c">

Shouldn't the `aria-placeholder` attribute change dynamically as users move between placeholder-enabled elements? Would a screen reader even respect that?

Reply:
Placeholder should provide an additional description for the field that is currently focused. Reading placeholders of all fields in one place defeats the purpose of that mechanism.

I would simply leave `aria-placeholder`s in the title and body fields, removing it from the editor itself. But I'm not sure if it would be read :thinking: Dynamically changing placeholder also sounds like something that would be ignored.

Reply:
@Comandeer / @oleq So what should we do?